python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+from getpass import getuser
+from time import time
+from typing import (
+    List,
+    Optional,
+    Set,
+    Union,
+)
+from urllib.parse import urlparse
+
+from hive_metastore.ThriftHiveMetastore import Client
+from hive_metastore.ttypes import AlreadyExistsException
+from hive_metastore.ttypes import Database as HiveDatabase
+from hive_metastore.ttypes import (
+    FieldSchema,
+    InvalidOperationException,
+    MetaException,
+    NoSuchObjectException,
+    SerDeInfo,
+    StorageDescriptor,
+)
+from hive_metastore.ttypes import Table as HiveTable
+from thrift.protocol import TBinaryProtocol
+from thrift.transport import TSocket, TTransport
+
+from pyiceberg.catalog import Identifier, Properties
+from pyiceberg.catalog.base import Catalog
+from pyiceberg.exceptions import (
+    AlreadyExistsError,
+    NamespaceNotEmptyError,
+    NoSuchNamespaceError,
+    NoSuchTableError,
+)
+from pyiceberg.schema import Schema
+from pyiceberg.table.base import Table
+from pyiceberg.table.partitioning import PartitionSpec
+from pyiceberg.types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IcebergType,
+    IntegerType,
+    ListType,
+    LongType,
+    MapType,
+    StringType,
+    StructType,
+    TimestampType,
+    TimeType,
+    UUIDType,
+)
+
+hive_types = {
+    BooleanType: "boolean",
+    IntegerType: "int",
+    LongType: "bigint",
+    FloatType: "float",
+    DoubleType: "double",
+    DateType: "date",
+    TimeType: "string",
+    TimestampType: "timestamp",
+    StringType: "string",
+    UUIDType: "string",
+    BinaryType: "binary",
+    FixedType: "binary",
+    DecimalType: None,
+    StructType: None,
+    ListType: None,
+    MapType: None,
+}
+
+
+class _HiveClient:
```

Comment:
I know we've been evaluating Python HMS clients. If any of this code is copied or based on one of the ones that we looked at, we need to credit the author and update our LICENSE and NOTICE files.

Reply:
Several libraries do it this way (for example [hive-metastore-client](https://github.com/quintoandar/hive-metastore-client)), but I wrote the code myself. This is the common way to handle context in Python, so not sure if stating a credit is necessary. If I hadn't looked at the other libraries, I would have done it in the same way.

Reply:
Sounds good. I just want to make sure we document where needed. No need since this is your work!